test/test_grid.py

```diff
@@ -534,9 +509,225 @@ def test_edge_nodes_euler(self):
+        self.assertEqual(mesh2_face_edges.sizes["nMesh2_face"],
+                         mesh2_face_nodes.sizes["nMesh2_face"])
+        self.assertEqual(mesh2_face_edges.sizes["nMaxMesh2_face_edges"],
+                         mesh2_face_nodes.sizes["nMaxMesh2_face_nodes"])
+
+        # Assert if the mesh2_edge_nodes sizes are correct.
+        # Euler formular for determining the edge numbers: n_face = n_edges - n_nodes + 2
+        num_edges = mesh2_face_edges.sizes["nMesh2_face"] + tgrid.ds[
+            "Mesh2_node_x"].sizes["nMesh2_node"] - 2
+        size = mesh2_edge_nodes.sizes["nMesh2_edge"]
+        self.assertEqual(mesh2_edge_nodes.sizes["nMesh2_edge"], num_edges)
+
+    def test_build_face_edges_connectivity_fillvalues(self):
+        f0_deg = [[120, -20], [130, -10], [120, 0], [105, 0], [95, -10],
+                  [105, -20]]
+        f1_deg = [[120, 0], [120, 10], [115, 0],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE]]
+        f2_deg = [[115, 0], [120, 10], [100, 10], [105, 0],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE]]
+        f3_deg = [[95, -10], [105, 0], [95, 30], [80, 30], [70, 0], [75, -10]]
+        f4_deg = [[65, -20], [75, -10], [70, 0], [55, 0], [45, -10], [55, -20]]
+        f5_deg = [[70, 0], [80, 30], [70, 30], [60, 0],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE]]
+        f6_deg = [[60, 0], [70, 30], [40, 30], [45, 0],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE],
+                  [ux.INT_FILL_VALUE, ux.INT_FILL_VALUE]]
+
+        verts = [f0_deg, f1_deg, f2_deg, f3_deg, f4_deg, f5_deg, f6_deg]
+        uds = ux.Grid(verts)
+        uds._build_face_edges_connectivity()
+        n_face = len(uds.ds["Mesh2_face_edges"].values)
+        n_node = uds.nMesh2_node
+        n_edge = len(uds.ds["Mesh2_edge_nodes"].values)
+
+        self.assertEqual(7, n_face)
+        self.assertEqual(21, n_node)
+        self.assertEqual(28, n_edge)
+
+        # We will utilize the edge_nodes_connectivity and face_edges_connectivity to generate the
+        # res_face_nodes_connectivity and compare it with the uds.ds["Mesh2_face_nodes"].values
+        edge_nodes_connectivity = uds.ds["Mesh2_edge_nodes"].values
+        face_edges_connectivity = uds.ds["Mesh2_face_edges"].values
+        face_nodes_connectivity = uds.ds["Mesh2_face_nodes"].values
+
+        # Create a dictionary to store the face indices for each edge
+        face_nodes_dict = {}
+
+        # Loop through each face and edge to build the dictionary
+        for face_idx, face_edges in enumerate(face_edges_connectivity):
+            for edge_idx in face_edges:
+                if edge_idx != ux.INT_FILL_VALUE or np.isnan(edge_idx):
```

Comment:
Including support for `nan` values is a precautionary measure in case users prefer to use `nan` instead of the default values. This feature does not cause any harm and may potentially be beneficial in the future.

Reply:
Internally we are always using integers for our connectivity arrays. It should never contain any nans